src/MultiMapManager/SafeParent.js

```diff
@@ -18,29 +18,34 @@ export default class SafeParent extends React.Component {
     this.ref = React.createRef()
   }
 
-  componentDidMount () {
-    const { Component, providerProps } = this.props
+  contextKeyLookup () {
+    const { Component, providerProps, inlineProps } = this.props
     const keys = Object.keys(providerProps)
+    const { current } = this.ref
 
-    if (this.ref.current) {
-      const parentContextKey = keys.find(key => {
-        const parentMap = this.ref.current.closest(`#${key}`)
+    if (current) {
+      if (inlineProps._ol_kit_context_id && current.parentElement?.classList && !current.parentElement?.classList?.contains(inlineProps._ol_kit_context_id)) {
+        current.parentElement.classList.add(inlineProps._ol_kit_context_id)
+        this.setState({ parentContextKey: inlineProps._ol_kit_context_id })
```

Comment:
@PatchesMaps does this actually work? how is the parent going to contain the `_ol_kit_context_id` in the class list??